src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -31,105 +31,104 @@
 				"and will be removed in the future");
 
-		damageables = exprs[0];
-		if (!LivingEntity.class.isAssignableFrom(damageables.getReturnType())) {
-			if (!ChangerUtils.acceptsChange(damageables, ChangeMode.SET, ItemType.class)) {
-				Skript.error(damageables + " cannot be changed, thus it cannot be damaged or repaired.");
-				return false;
-			}
-		}
-		damage = (Expression<Number>) exprs[1];
-		heal = matchedPattern >= 1;
-
+		this.damageables = exprs[0];
+		this.isHealing = matchedPattern >= 1;
+		this.amount = (Expression<Number>) exprs[1];
 		return true;
 	}
 
 	@Override
-	public void execute(Event e) {
-		double damage = 0;
-		if (this.damage != null) {
-			Number number = this.damage.getSingle(e);
-			if (number == null)
+	protected void execute(Event event) {
+		double amount = 0;
+		if (this.amount != null) {
+			Number amountPostCheck = this.amount.getSingle(event);
+			if (amountPostCheck == null)
 				return;
-			damage = number.doubleValue();
+			amount = amountPostCheck.doubleValue();
 		}
-		Object[] array = damageables.getArray(e);
-		Object[] newArray = new Object[array.length];
 
-		boolean requiresChange = false;
-		for (int i = 0; i < array.length; i++) {
-			Object value = array[i];
-			if (value instanceof ItemType) {
-				ItemType itemType = (ItemType) value;
-				ItemStack itemStack = itemType.getRandom();
+		for (Object obj : this.damageables.getArray(event)) {
+			if (obj instanceof ItemType) {
+				ItemType itemType = (ItemType) obj;
 
-				if (this.damage == null) {
+				if (this.amount == null) {
+					ItemUtils.setDamage(itemType, 0);
+				} else {
+					ItemUtils.setDamage(itemType, (int) Math2.fit(0, (ItemUtils.getDamage(itemType) + (isHealing ? -amount : amount)), itemType.getMaterial().getMaxDurability()));
+				}
+
+			} else if (obj instanceof Slot) {
+				Slot slot = (Slot) obj;
+				ItemStack itemStack = slot.getItem();
+
+				if (itemStack == null)
+					continue;
+
+				if (this.amount == null) {
 					ItemUtils.setDamage(itemStack, 0);
 				} else {
-					ItemUtils.setDamage(itemStack, (int) Math2.fit(0, ItemUtils.getDamage(itemStack) + (heal ? -damage : damage), itemStack.getType().getMaxDurability()));
+					ItemUtils.setDamage(itemStack, (int) Math2.fit(0, (ItemUtils.getDamage(itemStack) + (isHealing ? -amount : amount)), itemStack.getType().getMaxDurability()));
 				}
 
-				newArray[i] = new ItemType(itemStack);
-				requiresChange = true;
-			} else {
-				LivingEntity livingEntity = (LivingEntity) value;
-				if (!heal) {
-					HealthUtils.damage(livingEntity, damage);
-				} else if (this.damage == null) {
-					HealthUtils.setHealth(livingEntity, HealthUtils.getMaxHealth(livingEntity));
+				slot.setItem(itemStack);
+
+			} else if (obj instanceof Damageable) {
+				Damageable damageable = (Damageable) obj;
+
+				if (this.amount == null) {
+					HealthUtils.heal(damageable, HealthUtils.getMaxHealth(damageable) * 2);
```

Comment:
Why multiplying by 2?

Reply:
Actually something I don't need, was mostly added due to Skript's love of `/2` but it turns out `HealthUtils#setHealth` already multiples by 2.